lib/AST/ASTImporter.cpp

```diff
@@ -1994,7 +1994,7 @@ Decl *ASTNodeImporter::VisitRecordDecl(RecordDecl *D) {
     SearchName = Importer.Import(D->getTypedefNameForAnonDecl()->getDeclName());
     IDNS |= Decl::IDNS_Ordinary;
   } else if (Importer.getToContext().getLangOpts().CPlusPlus)
-    IDNS |= Decl::IDNS_Ordinary;
+    IDNS |= Decl::IDNS_Ordinary | Decl::IDNS_TagFriend;
```

Comment:
Perhaps we should handle the friend case in `VisitFunctionDecl` as well (`IDNS_OrdinaryFriend`) ?

Reply:
Yeah, that might be true. I am not sure about adding it without having an error that can be traced back to that.